### DISCUSSION TITLE
Get PNG as np array?

### DISCUSSION THREAD

**Discussion by lcgd240** [cduck/drawSvg#119] [Q&A] [answered]
Hi. I was wondering if there's a way of getting the drawing as a PNG, in three or four channels (RGB or RGBA) and then storing the values as an np array. It would be useful for those of us that also use OpenCV in conjunction with python. Many thanks!

**Comment by cduck** [comment, marked as answer]
Yes, you can use any Python library that can load PNG images like this: `imread(d.rasterize().png_data)`

Here's a demo:
```python
import drawsvg as draw
from imageio.v3 import imread, imwrite
from io import BytesIO

# Helper function
def display_array(img_array):
    '''Displays a numpy array as an image in Jupyter.'''
    with BytesIO() as f:
        imwrite(f, img_array, extension='.png')
        f.seek(0)
        return draw.Raster(f.read())
```
```python
# Make your drawing
d = draw.Drawing(100, 100)
d.append(draw.Circle(50, 50, 40, fill='red'))

d
```
![Drawing after cell 1](https://github.com/cduck/drawsvg/assets/2476062/c564f6d6-a677-49c2-97ef-b82e3600281b)
```python
# Convert your drawing into a numpy array
# The array shape is (height, width, 3) if opaque else (height, width, 4)
png_data = d.rasterize().png_data
img_array = imread(png_data, extension='.png')

display_array(img_array)
```
![Drawing after cell 2](https://github.com/cduck/drawsvg/assets/2476062/03863605-d0da-42f9-942f-e11216be82a2)
```python
# Manipulate your numpy array
img_array[::5, :] = (0, 255, 0, 255)
img_array[:, ::33] = (0, 255, 0, 255)

display_array(img_array)
```
![Drawing after cell 3](https://github.com/cduck/drawsvg/assets/2476062/1b611b54-f301-48b3-8fdf-a39a5e0799df)

**Reply by lcgd240** [reply]
Many thanks for the answer and the library! its amazingly powerful and easy to use :-)
